--- a/src/amqp_plugin.h
+++ b/src/amqp_plugin.h
@@ -21,6 +21,7 @@
 
 /* includes */
 #include <amqp.h>
+#include <amqp_tcp_socket.h>
 #include <sys/poll.h>
 
 /* structures */
--- a/src/sfv5_module.h
+++ b/src/sfv5_module.h
@@ -47,6 +47,7 @@
 EXT void sfv5_modules_db_init();
 EXT struct sfv5_modules_db_field *sfv5_modules_db_get_ie(u_int32_t);
 EXT struct sfv5_modules_db_field *sfv5_modules_get_next_ie(u_int32_t);
+EXT struct sfv5_modules_db_field *sfv5_modules_db_get_next_ie(u_int32_t);
 
 EXT struct sfv5_modules_desc sfv5_modules;
 #undef EXT
